lib/streams/recordStream.js

```diff
@@ -8,6 +9,29 @@ var CleanupStream = require('./cleanupStream');
 var ValidRecordFilterStream = require('./validRecordFilterStream');
 var DocumentStream = require('./documentStream');
 
+/*
+ * Construct a suitable id prefix for a CSV file given
+ * its full filename and the base directory of all OA CSV files.
+ */
+function getIdPrefix(filename, dirPath) {
+  var heirarchy = [];
+  var current_level = filename;
+
+  // traverse the directory structure upward to the OA base directory,
+  // keeping track of directory names along the way
+  while (current_level !== '/') {
```

Comment:
Ok, I like that. It definitely gives up any cross platform support, but we can come back and address that later (I'm sure it's not the only cross platform issue). I've updated this PR with code adapted from yours.